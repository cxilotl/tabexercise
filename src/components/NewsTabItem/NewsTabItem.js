import React from 'react';
import PropTypes from 'prop-types';
import cssStyles from './NewsTabItem.module.scss';

const NewsTabItem = ({ title, description, linkURL }) => {
  return (
    <article className={ cssStyles.layout }>
      <h4><a href={ linkURL } target="_blank" rel="noopener noreferrer">{ title }</a></h4>
      <p>{ description }</p>
    </article>
  );
};

NewsTabItem.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
};

NewsTabItem.defaultProps = {
  title: '',
  description: ''
};

export default NewsTabItem;