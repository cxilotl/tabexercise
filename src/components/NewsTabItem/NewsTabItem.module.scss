.layout {
  border: solid 1px blue;
}