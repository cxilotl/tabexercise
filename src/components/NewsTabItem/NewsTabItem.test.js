import React from 'react';
import renderer from 'react-test-renderer';
import NewsTabItem from './NewsTabItem';

describe('NewsTabItem', () => {
  const title = 'VAT ruling on Times digital edition could save News UK millions';
  const description = 'Rupert Murdochâ€™s firm satisfies tax tribunal that subscriptions should be zero-rated';
  const linkURL = 'https://www.theguardian.com/media/2020/jan/06/vat-ruling-on-times-digital-edition-could-save-news-uk-millions';

  it('Snapshot - Default', () => {
    const component = renderer.create(
      <NewsTabItem
        title={ title }
        description={ description }
        linkURL={ linkURL }
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});