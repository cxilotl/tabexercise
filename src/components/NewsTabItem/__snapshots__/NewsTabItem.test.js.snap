// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsTabItem Snapshot - Default 1`] = `
<article
  className="layout"
>
  <h4>
    <a
      href="https://www.theguardian.com/media/2020/jan/06/vat-ruling-on-times-digital-edition-could-save-news-uk-millions"
      rel="noopener noreferrer"
      target="_blank"
    >
      VAT ruling on Times digital edition could save News UK millions
    </a>
  </h4>
  <p>
    Rupert Murdochâ€™s firm satisfies tax tribunal that subscriptions should be zero-rated
  </p>
</article>
`;
